src/app/account/login/login.component.html

```diff
@@ -7,13 +7,32 @@
         <div class="login-card">
             <mat-card>
                 <mat-card-content>
+                    <p>
+                        <mat-error *ngIf="showError">{{errorMessage}}</mat-error>
+                    </p>
                     <h2>Welcome to iReflect</h2>
+                    <form [formGroup]="loginGroup" (ngSubmit)="onSubmit()">
+                        <mat-form-field appearance="outline">
+                          <input matInput placeholder="Email" formControlName="email" >
+                          <mat-error *ngIf="userEmail.hasError('email') && !userEmail.hasError('required')">
+                            Please enter a valid email address
+                          </mat-error>
+                          <mat-error *ngIf="userEmail.hasError('required')">
+                            Email is <strong>required</strong>
+                          </mat-error>
+                        </mat-form-field>
+                        <mat-form-field appearance="outline">
+                            <input matInput placeholder="Password" [type]="hidePassword ? 'password' : 'text'" formControlName="password" >
+                            <mat-icon matSuffix (click)="hidePassword = !hidePassword">{{!hidePassword ? 'visibility' : 'visibility_off'}}</mat-icon>
```

Comment:
Let's use positive if-else.
```suggestion
                            <mat-icon matSuffix (click)="hidePassword = !hidePassword">{{hidePassword ? 'visibility_off': 'visibility'}}</mat-icon>
```

Reply:
done.